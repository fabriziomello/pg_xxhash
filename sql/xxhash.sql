-- xxhash regression tests
\set VERBOSITY terse
\pset null 'NULL'
CREATE EXTENSION pg_xxhash;

-- XXH64 Tests
-- All tests below should return NULL
SELECT pg_xxhash64(NULL);
SELECT pg_xxhash64(NULL, NULL);
SELECT pg_xxhash64('foo', NULL);
SELECT pg_xxhash64(NULL, 0);

-- Testing using default seed = 0
SELECT pg_xxhash64('foo'); -- default seed = 0
SELECT pg_xxhash64('foo', 0);
SELECT pg_xxhash64('foo') = pg_xxhash64('foo', 0); -- should be TRUE
SELECT pg_xxhash64('foo') = pg_xxhash64('foo', 1); -- should be FALSE
SELECT pg_xxhash64('foo') = pg_xxhash64('foo', NULL); -- should be NULL
SELECT pg_xxhash64('foo') IS DISTINCT FROM pg_xxhash64('foo', NULL); -- should be TRUE

-- Testing using an specific seeds
SELECT pg_xxhash64('foo', 100);
SELECT pg_xxhash64('foo', -100);
SELECT pg_xxhash64('foo', 100) = pg_xxhash64('foo', -100); -- should be FALSE

-- XXH3_64bits tests
SELECT pg_xxh3_64(NULL); -- should return NULL
SELECT pg_xxh3_64('foo');
SELECT pg_xxh3_64('bar');
SELECT pg_xxh3_64('foo') = pg_xxh3_64('foo'); -- should be TRUE
SELECT pg_xxh3_64('foo') = pg_xxh3_64('bar'); -- should be FALSE

DROP EXTENSION pg_xxhash;
