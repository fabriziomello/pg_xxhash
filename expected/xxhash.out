-- xxhash regression tests
\set VERBOSITY terse
\pset null 'NULL'
CREATE EXTENSION pg_xxhash;
-- XXH64 Tests
-- All tests below should return NULL
SELECT pg_xxhash64(NULL);
 pg_xxhash64 
-------------
        NULL
(1 row)

SELECT pg_xxhash64(NULL, NULL);
 pg_xxhash64 
-------------
        NULL
(1 row)

SELECT pg_xxhash64('foo', NULL);
 pg_xxhash64 
-------------
        NULL
(1 row)

SELECT pg_xxhash64(NULL, 0);
 pg_xxhash64 
-------------
        NULL
(1 row)

-- Testing using default seed = 0
SELECT pg_xxhash64('foo'); -- default seed = 0
     pg_xxhash64     
---------------------
 3728699739546630719
(1 row)

SELECT pg_xxhash64('foo', 0);
     pg_xxhash64     
---------------------
 3728699739546630719
(1 row)

SELECT pg_xxhash64('foo') = pg_xxhash64('foo', 0); -- should be TRUE
 ?column? 
----------
 t
(1 row)

SELECT pg_xxhash64('foo') = pg_xxhash64('foo', 1); -- should be FALSE
 ?column? 
----------
 f
(1 row)

SELECT pg_xxhash64('foo') = pg_xxhash64('foo', NULL); -- should be NULL
 ?column? 
----------
 NULL
(1 row)

SELECT pg_xxhash64('foo') IS DISTINCT FROM pg_xxhash64('foo', NULL); -- should be TRUE
 ?column? 
----------
 t
(1 row)

-- Testing using an specific seeds
SELECT pg_xxhash64('foo', 100);
     pg_xxhash64     
---------------------
 4679220945261188543
(1 row)

SELECT pg_xxhash64('foo', -100);
     pg_xxhash64     
---------------------
 4918571626677971423
(1 row)

SELECT pg_xxhash64('foo', 100) = pg_xxhash64('foo', -100); -- should be FALSE
 ?column? 
----------
 f
(1 row)

-- XXH3_64bits tests
SELECT pg_xxh3_64(NULL); -- should return NULL
 pg_xxh3_64 
------------
       NULL
(1 row)

SELECT pg_xxh3_64('foo');
      pg_xxh3_64      
----------------------
 -6093828362558603894
(1 row)

SELECT pg_xxh3_64('bar');
      pg_xxh3_64      
----------------------
 -3142447797644373150
(1 row)

SELECT pg_xxh3_64('foo') = pg_xxh3_64('foo'); -- should be TRUE
 ?column? 
----------
 t
(1 row)

SELECT pg_xxh3_64('foo') = pg_xxh3_64('bar'); -- should be FALSE
 ?column? 
----------
 f
(1 row)

DROP EXTENSION pg_xxhash;
